src/flow/maximum_flow.jl

```diff
@@ -59,16 +61,16 @@ Requires arguments:
 - flow_graph::DiGraph,                    # the input graph
 - capacity_matrix::AbstractArray{T,2}     # input capacity matrix
 """
-
-residual(flow_graph::DiGraph) = DiGraph(Graph(flow_graph))
+function residual end
```

Comment:
docstrings cannot precede `@traitfn`s. You'll see dummy methods sprinkled throughout. The other way to do this (if this causes docgen problems) is to put all the docs at the bottom like so:

```
@traitfn foo{G<:AbstractGraph; IsDirected{G}}(g::G) = rand(vertices(g))

"""
foo(g::DiGraph)
Returns some silly number.
"""
foo
```